fennel/datasets/test_schema_validator.py

```diff
@@ -1510,42 +1510,6 @@ def pipeline_window(cls, app_event: Dataset):
     )
 
 
-def test_join_with_wrong_right_index():
```

Comment:
Instead of removing it, can you instead modify it to confirm that it's possible to join with a dataset without the right index?